res/controllers/Denon-MC7000-scripts.js

```diff
@@ -837,6 +837,20 @@ MC7000.censor = function(channel, control, value, status, group) {
     }
 };
 
+// Auto-Loop Button
+MC7000.autoLoop = function(channel, control, value, status, group) {
+    // TODO: handle shift button
+    if (value) { // Button Down == 0x7F
+        if (engine.getValue(group, "loop_enabled")) {
+            engine.setValue(group, "loop_enabled", false);
+            engine.setValue(group, "reloop_toggle", true);
+        } else {
+            engine.setValue(group, "beatloop_activate", true);
```

Comment:
This invocation sequence requires a comment, i.e. why you need a negative edge trigger. Does this even work as intended?

Reply:
This comment is imprecise and wrong. script.toggleControl() works as expected and inverts a boolean value, i.e. 0/false -> 1/true or 1/true -> 0/false. Setting `beatloop_activate` to `true` and then to `false` is a completely different use case.

How about setting `beatloop_activate` to `true` on *button down* and to `false` otherwise (*button up* or when entering the other `if` arm on *button down*)?

I am using this pattern a lot in the MC6000MK2 script to avoid getting stuck in inconsistent states, e.g. when a button is pressed *shifted* and released *unshifted* or the other way round. All control values that might be affected by a button and all invocation sequences must be considered.